### PR TITLE
Added name checking

### LINKED ISSUE #8933: Tell certificate earners how to add their names
Campers will often claim a certificate, then not understand why their certificate doesn't have their name on it. This is because we pull their name from GitHub, and if they haven't added a name to their GitHub account, we don't have their name, either.

Agile user story:
- As a user, if I attempt to claim a certificate and my account doesn't have a name, I should see instructions for how to add it.
- I should not be able to claim a certificate while my name is blank

Copy for the flash message: 

"We need your name so we can put it on your certificate. [Add your name to your GitHub account](https://github.com/settings/profile), then go to your [settings page](https://www.freecodecamp.com/settings) and click the "update my portfolio from GitHub" button. Then we can issue your certificate."

### PR DESCRIPTION
**Pull request by Trion129**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Git-Squash) them into one commit).
- [x] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)
#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue: Closes #8933 
#### Description

Added functionality for checking if name is present for the user, if user has already claimed the certificate the certificate will show a link to his github profile. While claiming certificate if the user has no name provided, the user is shown a message to do that and user is not allowed to claim the certificate until a name is provided

### CODE REVIEW

File: seed/challenges/01-front-end-development-certification/front-end-development-certificate.json [hunk 1]
```diff
@@ -20,6 +20,12 @@
           "#"
         ],
         [
+          "//i.imgur.com/TDT6vJ1.jpg",
+          "Your github name is used as the name on your certificates",
+          "Let's confirm that you have your github name set up and imported it on freecodecamp. If not, you will be redirected to your freecodecamp profile from where you can import your github name",
```
Comment (bugron) [inline seed/challenges/01-front-end-development-certification/front-end-development-certificate.json]: `freecodecamp` must be `Free Code Camp` everywhere.

File: server/utils/middleware.js [hunk 2]
```diff
@@ -29,6 +29,18 @@ export function ifNoUser401(req, res, next) {
   return res.status(401).end();
 }
 
+export function isNamed(req, res, next) {
+  if (req.user.name === '') {
+    return res.status(200).send(`We need your name so we can put it on your
+      certificate.<a href="https://github.com/settings/profile"> Add your name
+      to your GitHub account</a>, then
+      go to your <a href="https://www.freecodecamp.com/settings">
+      settings</a> page and click the "update my portfolio from GitHub" button.
```
Comment (bugron) [inline server/utils/middleware.js]: Make `u` uppercase: `Update my...`.

File: server/utils/middleware.js [hunk 3]
```diff
@@ -29,6 +29,18 @@ export function ifNoUser401(req, res, next) {
   return res.status(401).end();
 }
 
+export function isNamed(req, res, next) {
+  if (req.user.name === '') {
+    return res.status(200).send(`We need your name so we can put it on your
+      certificate.<a href="https://github.com/settings/profile"> Add your name
```
Comment (bugron) [inline server/utils/middleware.js]: You have a space before the word `Add`. Move it before the `a` tag:
`...certificate. <a href="https://github.com/settings/profile">Add your nam...`

**Comment by bugron** [comment]
@FreeCodeCamp/issue-moderators here is what it looks like:
Pic. 1

Pic. 2

Pic. 3

**Comment by raisedadead** [comment]
For step 1 (Pic 1)
- A more logical way should be if a user has already claimed a certificate and doesn't have a name on it then we should push a notification alert asking them to do so.
- Other than that, the certificate view should not be rendered at all (with the same notificaton as above).

The next steps (Pic 2 and 3) LGTM!

**Comment by Trion129** [comment]
:smile: Added the flash message as @raisedadead suggested and made all the necessary changes in the json file

**Comment by Trion129** [comment]
Please check the grammar of french and spanish translations, I used google translate to get a placeholder translation so not very confident about them

File: server/views/certificate/script.jade [hunk 4]
```diff
@@ -2,6 +2,12 @@ script.
   (function() {
     var containerWidth = document.getElementById('cert').offsetWidth;
     var nameDiv = document.getElementById('name');
+    if(nameDiv.innerHTML == ""){
```
Comment (bugron) [inline server/views/certificate/script.jade]: Is this block still necessary?

**Comment by bugron** [comment]
Trying to access user's certification route redirects back to the challenge:
Pic 1. - v2.0

File: server/utils/middleware.js [hunk 5]
```diff
@@ -29,6 +29,18 @@ export function ifNoUser401(req, res, next) {
   return res.status(401).end();
 }
 
+export function isNamed(req, res, next) {
+  if (req.user.name === '') {
+    return res.status(200).send(`We need your name so we can put it on your
```
Comment (bugron) [inline server/utils/middleware.js]: Feels like this string wants to be `dedent`ified too:

File: seed/challenges/01-front-end-development-certification/front-end-development-certificate.json [hunk 6]
```diff
@@ -254,6 +263,12 @@
           "#"
         ],
         [
+          "//i.imgur.com/TDT6vJ1.jpg",
+          "Su nombre github se utiliza como el nombre de sus certificados",
```
Comment (vtamara) [inline seed/challenges/01-front-end-development-certification/front-end-development-certificate.json]: Better "Su nombre en github se utiliza como nombre en sus certificados"

File: seed/challenges/01-front-end-development-certification/front-end-development-certificate.json [hunk 7]
```diff
@@ -254,6 +263,12 @@
...
+          "Vamos a confirmar que ha establecido su nombre github y importada en Free Code Camp gratuito. Si no es así, se le dará instrucciones para importar a su perfil de Free Code Camp. Usted no puede reclamar el certificado hasta que proporcione un nombre.",
```
Comment (vtamara) [inline seed/challenges/01-front-end-development-certification/front-end-development-certificate.json]: Better "Vamos a confirmar que has establecido tu nombre en github y que fue importado a Free Code Camp. Si no es así, se te darán instrucciones para que lo importes a tu perfil de Free Code Camp. No podrás reclamar el certificado sino hasta que proporciones un nombre."

File: seed/challenges/02-data-visualization-certification/data-visualization-certificate.json [hunk 8]
```diff
@@ -106,6 +115,12 @@
           "#"
         ],
         [
+          "//i.imgur.com/TDT6vJ1.jpg",
+          "Su nombre github se utiliza como el nombre de sus certificados",
```
Comment (vtamara) [inline seed/challenges/02-data-visualization-certification/data-visualization-certificate.json]: Better "Tu nombre en github se utiliza como nombre en tus certificados"

File: seed/challenges/03-back-end-development-certification/back-end-development-certificate.json [hunk 9]
```diff
@@ -106,6 +115,12 @@
           "#"
         ],
         [
+          "//i.imgur.com/TDT6vJ1.jpg",
+          "Su nombre github se utiliza como el nombre de sus certificados",
```
Comment (vtamara) [inline seed/challenges/03-back-end-development-certification/back-end-development-certificate.json]: Better "Tu nombre en github se utiliza como nombre en tus certificados"

File: seed/challenges/03-back-end-development-certification/back-end-development-certificate.json [hunk 10]
```diff
@@ -106,6 +115,12 @@
...
+          "Vamos a confirmar que ha establecido su nombre github y importada en Free Code Camp gratuito. Si no es así, se le dará instrucciones para importar a su perfil de Free Code Camp. Usted no puede reclamar el certificado hasta que proporcione un nombre.",
```
Comment (vtamara) [inline seed/challenges/03-back-end-development-certification/back-end-development-certificate.json]: Better "Vamos a confirmar que has establecido tu nombre en github y que fue importado a Free Code Camp. Si no es así, se te darán instrucciones para que lo importes a tu perfil de Free Code Camp. No podrás reclamar el certificado sino hasta que proporciones un nombre."

**Comment by raisedadead** [comment]
@trion can you re-base this?

**Comment by Trion129** [comment]
@raisedadead Rebased! :)

**Comment by raisedadead** [comment]
Mods:
### DO NOT MERGE YET

File: server/boot/certificate.js [hunk 11]
```diff
@@ -149,6 +150,13 @@ export default function certificate(app) {
     [certTypes.backEnd]: getIdsForCert$(backEndChallengeId, Challenge)
   };
 
+
+  router.post(
+    '/certificate/verify/name',
+    ifNoUser401,
+    isNamed
```
Comment (BerkeleyTrue) [inline server/boot/certificate.js]: There is no route handler here

**Comment by raisedadead** [comment]
@Trion129 closing as stale, please re-open once ready.

**Pull request closed without merging**